plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Type.hs

```diff
@@ -80,6 +81,11 @@ data Term name uni fun ann
     -- This is the cutoff at which constructors won't get pointer tags
     -- See Note [Term constructor ordering and numbers]
     | Error !ann
+    -- TODO: worry about overflow, maybe use an Integer
+    -- TODO: try spine-strict list or strict list or vector
+    -- See Note [Constr tag type]
+    | Constr !ann !Word64 ![Term name uni fun ann]
+    | Case !ann !(Term name uni fun ann) ![Term name uni fun ann]
```

Comment:
This look like it should be a vector or something. At the very least a strict list.